### PR TITLE
Refactor out the StepManager class

### PR DESCRIPTION
**Pull request by whoojemaflip**
Moves the responsibility for managing URL state into the controller. This change does away with the need to mix that stateful representation into our Cells and our Forms. It's provided here as a template for further work in the area.

### CODE REVIEW

File: app/controllers/healthcare_controller.rb [hunk 1]
```diff
@@ -1,44 +1,81 @@
...
-    end
-  end
+  private
 
-  def add_medication
-    step.form
-    step.form.deserialize(params[step_name])
-    step.form.add_medication
+  def healthcare_form_cell
+    cell = cell(:healthcare, form)
```
Comment (dan-palmer) [inline app/controllers/healthcare_controller.rb:44]: Taking a sort of builder approach to this is quite neat

File: app/controllers/healthcare_controller.rb [hunk 2]
```diff
@@ -1,44 +1,81 @@
...
+    cell.current_question = current_step_index + 1
+    cell.total_questions = wizard_steps.size
+    cell.template_name = form.class.name
+    cell.form_path = wizard_path
+
+    cell
   end
 
-  private
+  def permitted_params
```
Comment (dan-palmer) [inline app/controllers/healthcare_controller.rb]: How about just having this as `params`, the permitting part should be handled by reform right?

File: app/models/forms/base.rb [hunk 3]
```diff
@@ -40,5 +40,26 @@ def invalid?
     def toggle_choices
       TOGGLE_CHOICES
     end
+
+    def to_parameter_hash
+      param_hash = {}
+
+      @fields.each do |k, v|
```
Comment (dan-palmer) [inline app/models/forms/base.rb]: How about `each_with_object` here?

**Comment by cesidio** [comment]
👍

**Pull request merged**